plutus-core/plutus-core/src/PlutusCore/Rename/Internal.hs

```diff
@@ -16,13 +16,15 @@ import           PlutusCore.Name
 import           PlutusCore.Quote
 import           PlutusCore.Rename.Monad as Export
 
+import           Control.Monad.Reader
+
 -- | Replace the unique in the name stored in a 'TyVarDecl' by a new unique, save the mapping
 -- from the old unique to the new one and supply the updated 'TyVarDecl' to a continuation.
 withFreshenedTyVarDecl
-    :: (HasRenaming ren TypeUnique, HasUniques (Type tyname uni ann), MonadQuote m)
+    :: (HasRenaming ren TypeUnique, HasUniques (Type tyname uni ann), MonadQuote m, MonadReader ren m)
     => TyVarDecl tyname ann
-    -> (TyVarDecl tyname ann -> RenameT ren m c)
-    -> RenameT ren m c
+    -> (TyVarDecl tyname ann -> m c)
+    -> m c
```

Comment:
Generalized stuff related to renaming, so that we can plug in flawed renamers and check that they cause tests to fail.